app/forms/section2/carrier/CarrierEoriNumber.scala

```diff
@@ -45,9 +45,8 @@ object CarrierEoriNumber extends DeclarationPage {
   def form: Form[CarrierEoriNumber] = Form(mapping)
 
   def apply(carrierDetails: CarrierDetails): CarrierEoriNumber =
-    carrierDetails.details.eori match {
-      case Some(eori) => CarrierEoriNumber(Some(eori), YesNoAnswers.yes)
-      case _          => CarrierEoriNumber(None, YesNoAnswers.no)
+    carrierDetails.details.eori.fold(CarrierEoriNumber(None, YesNoAnswers.no)) { eori =>
+      CarrierEoriNumber(Some(eori), YesNoAnswers.yes)
```

Comment:
Personally I still prefer the match syntax to the fold! :wink:

Reply:
With the pattern matching I would need to change the naming of `eori` to something different. This is why I went with the `fold`